### DISCUSSION TITLE
About formik performance

### DISCUSSION THREAD

**Discussion by onuralpbicer** [jaredpalmer/formik#3186] [Q&A] [answered]
So currently we're developing an app that is basically a page editor. Think Figma but way more specific to the industry that our company is in. User can drop "components"(not in the React sense) into a page and configure them. I had the idea to have the whole thing as a formik form, as in when the file is loaded, have formik handle the current value and changes etc. The current way of doing things is updating a redux store, which works fine but if I could have it as a formik form, it might make the code cleaner/better dx. So, people who are familiar with the library, does this sound like a good idea?

**Comment by johnrom** [comment, marked as answer]
Formik's current API isn't necessarily built for dynamic data like dropping in new components. It is heavily based around the concept of `initialValues` and updating the values within them. It's not impossible to work around this limitation, but it won't be super straightforward if you planned to use `touched` on your new fields. This is probably Formik's 2nd biggest limitation currently.

If it's really a page editor, I'd look to WordPress' Gutenberg for inspiration as they are really nailing it over there.

**Reply by onuralpbicer** [reply]
Yea thought as much, wouldn't mind working around that limitation, I don't think I'd use touched. My biggest concern is mostly about performance. Would changing one "field" re-render every component that is using setfieldvalue etc?

And it's not exactly that, I'll look into it but it's a highly highly customised thing for the industry

**Reply by johnrom** [reply]
Formik v2 does re-render every field. V3 does not, though I haven't heard anything about a release. I've just been using my own fork for a while.

**Reply by onuralpbicer** [reply]
Very interesting, then it might actually be a viable option.. The app is already mostly built at the moment, so it's not something I'm gonna attempt in the very near future but at some point during a refactor it's an option I might explore more.

Looking forward to v3, formik is one of my favorite React libraries to work with. Thanks for your input
